From a2e7d77518ba4f5637c0ddb4368de5005fc1d2c8 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Fri, 25 Sep 2020 02:01:31 -0700
Subject: [PATCH 1666/1852] ASoC: xlnx: formatter: Fix warnings for
 NULL_RETURNS

commit 56c86114520e5c281847df0d896a4a835dbda85c from
https://github.com/Xilinx/linux-xlnx.git

This patch fix the warnings for the NULL_RETURNS type.

The actual warnings for below functions are:
xlnx_formatter_pcm_new():
xlnx_formatter_pcm_open():
xlnx_formatter_pcm_hw_params():
xlnx_formatter_pcm_close():

Event returned_null: "snd_soc_rtdcom_lookup" returns "NULL".
Event var_assigned: Assigning: "component" = "NULL" return value
from "snd_soc_rtdcom_lookup".
Event dereference: Dereferencing "component", which is known to be
"NULL".

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 sound/soc/xilinx/xlnx_formatter_pcm.c | 21 +++++++++++++++++++--
 1 file changed, 19 insertions(+), 2 deletions(-)

diff --git a/sound/soc/xilinx/xlnx_formatter_pcm.c b/sound/soc/xilinx/xlnx_formatter_pcm.c
index ea136cd482ce..ad4bf5bb9ebc 100644
--- a/sound/soc/xilinx/xlnx_formatter_pcm.c
+++ b/sound/soc/xilinx/xlnx_formatter_pcm.c
@@ -333,7 +333,12 @@ static int xlnx_formatter_pcm_open(struct snd_pcm_substream *substream)
 	struct snd_soc_pcm_runtime *prtd = substream->private_data;
 	struct snd_soc_component *component = snd_soc_rtdcom_lookup(prtd,
 								    DRV_NAME);
-	struct xlnx_pcm_drv_data *adata = dev_get_drvdata(component->dev);
+	struct xlnx_pcm_drv_data *adata;
+
+	if (!component)
+		return -ENODEV;
+
+	adata = dev_get_drvdata(component->dev);
 
 	if (substream->stream == SNDRV_PCM_STREAM_PLAYBACK &&
 	    !adata->mm2s_presence)
@@ -406,6 +411,9 @@ static int xlnx_formatter_pcm_close(struct snd_pcm_substream *substream)
 	struct snd_soc_component *component = snd_soc_rtdcom_lookup(prtd,
 								    DRV_NAME);
 
+	if (!component)
+		return -ENODEV;
+
 	ret = xlnx_formatter_pcm_reset(stream_data->mmio);
 	if (ret) {
 		dev_err(component->dev, "audio formatter reset failed\n");
@@ -446,9 +454,14 @@ static int xlnx_formatter_pcm_hw_params(struct snd_pcm_substream *substream,
 	struct snd_soc_pcm_runtime *prtd = substream->private_data;
 	struct snd_soc_component *component = snd_soc_rtdcom_lookup(prtd,
 								    DRV_NAME);
-	struct xlnx_pcm_drv_data *adata = dev_get_drvdata(component->dev);
+	struct xlnx_pcm_drv_data *adata;
 	struct snd_soc_pcm_runtime *rtd = substream->private_data;
 
+	if (!component)
+		return -ENODEV;
+
+	adata = dev_get_drvdata(component->dev);
+
 	bits_per_sample = params_width(params);
 	sample_rate = params_rate(params);
 	active_ch = params_channels(params);
@@ -563,6 +576,10 @@ static int xlnx_formatter_pcm_new(struct snd_soc_pcm_runtime *rtd)
 {
 	struct snd_soc_component *component = snd_soc_rtdcom_lookup(rtd,
 								    DRV_NAME);
+
+	if (!component)
+		return -ENODEV;
+
 	snd_pcm_lib_preallocate_pages_for_all(rtd->pcm,
 			SNDRV_DMA_TYPE_DEV, component->dev,
 			xlnx_pcm_hardware.buffer_bytes_max,
-- 
2.31.1

